### PR TITLE
Feature/seismicity rate solver

### PR DESCRIPTION
**Pull request by kimtaeho07**

### CODE REVIEW

File: src/docs/doxygen/GeosxConfig.hpp
Comment (herve-gross) [inline src/docs/doxygen/GeosxConfig.hpp:1]: I don't think that this file should be part of the PR.

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+namespace dataRepository
+{
+namespace keys
+{}
+}
+
+using namespace dataRepository;
+using namespace fields;
+
+/*----------------------------------------------------------------------------------
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:34]: I would edit out everything related to the documentation of the Laplace solver here, and explain what this solver does instead.

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.hpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+#ifndef GEOS_PHYSICSSOLVERS_INDUCED_SEISMICITY_DIETERICH_SEISMICITY_RATE_HPP_
+#define GEOS_PHYSICSSOLVERS_INDUCED_SEISMICITY_DIETERICH_SEISMICITY_RATE_HPP_
+
+#include "physicsSolvers/inducedSeismicity/SeismicityRateBase.hpp"  // a base class shared by all Laplace solvers
+
+namespace geos
+{
+
+// Like most physics solvers, the Laplace solver derives from a generic SolverBase class.
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.hpp:23]: I would scrub the file from the Laplace-related comments.

File: src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+DECLARE_FIELD( seismicityRate,
+               "seismicityRate",
+               array1d< real64 >,
+               0,
+               LEVEL_0,
+               WRITE_AND_READ,
+               "Seismicity rate");
+
+DECLARE_FIELD( h,
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp:132]: I would use a longer more descriptive variable name, for readability (unless of course, everyone understands what "h" means in this context). One-letter variable names should be avoided when representing physical quantities.

File: src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+               "h",
+               array1d< real64 >,
+               0,
+               LEVEL_0,
+               WRITE_AND_READ,
+               "Log of the seismicity rate");
+
+DECLARE_FIELD( h_n,
```
Comment (herve-gross) [inline src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp:140]: Same remark here. Avoid one-letter variable names.

**Review by herve-gross — COMMENTED** [review]
A few minor remarks, mostly on form (I did not try to compile or test the new solver). You may be blocked by Doxygen or Uncrustify in the CI. Otherwise, this is a great start! Thanks.

File: src/docs/doxygen/GeosxConfig.hpp
Comment (kimtaeho07) [reply src/docs/doxygen/GeosxConfig.hpp:1]: Got it, will add it.

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp (hunk 1 above)
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:34]: Got it, I'll review what I might have left out in other places as well.

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.hpp (hunk 2 above)
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.hpp:23]: Will do.

File: src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp (hunk 3 above)
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp:132]: Will change to `logSeismicityRate`

File: src/coreComponents/LvArray
Comment (CusiniM) [inline src/coreComponents/LvArray:1]: I suspect that you forgot a `git submodule update `

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 5]
```diff
@@ -0,0 +1,323 @@
...
+      // solve for the seismcity rate given new stresses on faults
+      integralSolverStep( time_n, dtStress, subRegion ); 
+    });
+  });
+
+  // return time step size achieved by stress solver
+  return dtStress;
+}
+
+// Solve backward-Euler discretization of ODE by Newton-Raphson
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:149]: all this commented code should probably be removed. Or, if it works, I guess that we can clean it up and keep it in here.

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 6]
```diff
@@ -0,0 +1,323 @@
...
+  arrayView1d< real64 const > const sig_n = subRegion.getField< inducedSeismicity::meanNormalStress_n >();
+  
+  arrayView1d< real64 const > const tau_i = subRegion.getField< inducedSeismicity::initialMeanShearStress >();
+  arrayView1d< real64 const > const tau = subRegion.getField< inducedSeismicity::meanShearStress >();
+  arrayView1d< real64 const > const tau_n = subRegion.getField< inducedSeismicity::meanShearStress_n >();
+
+  // hard-coded parameter for testing solver against analytical solution 
+  // real64 c = 1e-3;
+
+  forAll< parallelDevicePolicy<> >(  h.size(), [=] GEOS_HOST_DEVICE ( localIndex const k )
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:249]:
```suggestion
  forAll< parallelDevicePolicy<> >(  subRegion.size(), [=] GEOS_HOST_DEVICE ( localIndex const k )
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+ * --------------------------
+ *
+ * This solver finds a solution R(x, t) - the seismicity rate - to the ODE formulated 
+ * by Dieterich, 1994 given a certain stressing history. The stressing history can consist
+ * of mechanical stresses and pore pressure. The solver class includes a member variable 
+ * pointing to the stress solver that is specified in the XML file. SolverStep for the 
+ * stress solver is then called in the SolverStep function for the seismicity rate, to take
+ * the updated stress history as the input.
+ * 
+ * Solving the ODE is currently implemented by computing the closed-form interal solution
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:50]:
```suggestion
 * Solving the ODE is currently implemented by computing the closed-form integral solution 
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+using namespace constitutive;
+
+/*----------------------------------------------------------------------------------
+ * DieterichSeismicityRate: Solving the ODE for seismicity rate from Dieterich, 1994
+ * ---------------------------------------------------------------------------------
+ *
+ * What does this solver do?
+ * --------------------------
+ *
+ * This solver finds a solution R(x, t) - the seismicity rate - to the ODE formulated
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:43]:
```suggestion
 * This solver finds a solution R(x, t) - the seismicity rate - to the ordinary differential equation (ODE) formulated 
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 9]
```diff
@@ -0,0 +1,323 @@
...
+   } );
+}
+//END_SPHINX_INCLUDE_REGISTERDATAONMESH
+
+real64 DieterichSeismicityRate::solverStep( real64 const & time_n,
+                                  real64 const & dt,
+                                  const int cycleNumber,
+                                  DomainPartition & domain )
+{
+  // Save initial stress state on pre-defined fault orienations to field variables
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:118]:
```suggestion
  // Save initial stress state on pre-defined fault orientations to field variables
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp [hunk 10]
```diff
@@ -0,0 +1,323 @@
...
+                                                [&]( localIndex const,
+                                                     ElementSubRegionBase & subRegion )
+    {
+      // project new stress state to update stress on fault
+      if ( subRegion.hasWrapper( SolidMechanicsLagrangianFEM::viewKeyStruct::solidMaterialNamesString() ) )
+      {
+        updateMeanSolidStress( subRegion );
+      }
+
+      // solve for the seismcity rate given new stresses on faults
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/DieterichSeismicityRate.cpp:140]:
```suggestion
      // solve for the seismicity rate given new stresses on faults
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+                                                              [&]( localIndex const,
+                                                                   ElementSubRegionBase & subRegion )
+    {
+      subRegion.registerField< inducedSeismicity::meanStress >( getName() ).
+        reference().resizeDimension< 1 >( 6 );
+
+      subRegion.registerField< inducedSeismicity::initialMeanNormalStress >( getName() );
+      subRegion.registerField< inducedSeismicity::initialMeanShearStress >( getName() );
+
+      subRegion.registerField< inducedSeismicity::meanNormalStress >( getName() );
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp:78]: As a general comment, it seems to me like we are potentially storing lots of data arrays that could be computed on the fly.  For example, could we just add a function to the solid material model to return the mean normal and shear stresses as needed?  For the seismicity rate calculation itself, are we storing any temporary quantities that could be recomputed on the fly, or do we need all these arrays (R, h, logDenom, etc)?

Comment (joshua-white) [reply src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp:78]: Also, perhaps we should discuss the terminology.  The normal stress here is the projected normal traction on the fault, averaged over quadrature points?  Perhaps we can call this `projectedNormalTraction` and `projectedShearTraction`?  At first I was thinking this was some measure of the volumetric and deviatory stress in the body.  If the nomenclature is clear to others, though, let's keep it.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp [hunk 12]
```diff
@@ -0,0 +1,257 @@
...
+
+    // Compute average stress state
+    for( int q = 0; q < stress.size(1); q++ ) 
+    {
+      LvArray::tensorOps::add< 6 >( meanStress[k], stress[k][q] );
+    }
+    LvArray::tensorOps::scale< 6 >(meanStress[k], 1./stress.size(1));
+
+    // Project average stress state to fault
+    sig[k] = LvArray::tensorOps::AiBi< 6 >( meanStress[k], m_faultNormalVoigt);
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp:142]: This is great if the user can provide a desired fault orientation.  Is it useful to add the option to find the "critical orientation"--that is, if the medium contained fractures in all direction, which orientation is most likely to be activated?  This would be a lower bound.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp [hunk 13]
```diff
@@ -0,0 +1,257 @@
...
+  {
+    if( !checkFaultOrthogonality() ){
+      GEOS_ERROR("Fault normal and fault shear must be orthogonal");
+    }
+
+    normalizeFaultVecs(); 
+
+    // Voigt notation of dyadic product of fault normal vectors 
+    // when multiplied in double dot product with symmetric stress tensor
+    m_faultNormalVoigt[0] = m_faultNormal[0]*m_faultNormal[0];
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp:228]: I might call these `m_faultNormalProjectionTensor` and `m_faultShearProjectionTensor` to emphasize there are tensors and not vectors.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp [hunk 14]
```diff
@@ -0,0 +1,257 @@
...
+    m_faultShearVoigt[0] = m_faultShear[0]*m_faultNormal[0];
+    m_faultShearVoigt[1] = m_faultShear[1]*m_faultNormal[1];
+    m_faultShearVoigt[2] = m_faultShear[2]*m_faultNormal[2];
+    m_faultShearVoigt[3] = m_faultShear[1]*m_faultNormal[2] + m_faultShear[2]*m_faultNormal[1];
+    m_faultShearVoigt[4] = m_faultShear[0]*m_faultNormal[2] + m_faultShear[2]*m_faultNormal[0];
+    m_faultShearVoigt[5] = m_faultShear[0]*m_faultNormal[1] + m_faultShear[1]*m_faultNormal[0];
+  }
+}
+
+void SeismicityRateBase::normalizeFaultVecs()
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp:246]: not sure we really need a separate function for this.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.hpp [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+  virtual ~SeismicityRateBase() override;
+
+  /// This method ties properties with their supporting mesh
+  virtual void registerDataOnMesh( Group & meshBodies ) override; 
+
+  struct viewKeyStruct : public SolverBase::viewKeyStruct
+  {
+    static constexpr char const * stressSolverNameString() { return "stressSolverName"; }
+    static constexpr char const * faultNormalString() { return "faultNormal"; }
+    static constexpr char const * faultShearString() { return "faultShear"; }
```
[the remark this answers is not in the document]
Comment (joshua-white) [reply src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.hpp:49]: Totally agree.  Also, a basis in 3D usually needs three vectors.  Are only two required?  This seems to fix the slip orientation.

File: src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp [hunk 16]
```diff
@@ -0,0 +1,176 @@
...
+#include "common/DataLayouts.hpp"
+#include "mesh/MeshFields.hpp"
+
+namespace geos
+{
+
+namespace fields
+{
+
+namespace inducedSeismicity
```
Comment (joshua-white) [inline src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp:31]: Perhaps all of these fields are required, but it is worth asking if any can be dropped and just recomputed as needed.

File: src/docs/doxygen/GeosxConfig.hpp [hunk 17]
```diff
@@ -30,7 +30,7 @@
 #define GEOSX_USE_MPI
 
 /// Enables use of OpenMP (CMake option ENABLE_OPENMP)
-#define GEOSX_USE_OPENMP
+/* #undef GEOSX_USE_OPENMP */
```
Comment (joshua-white) [inline src/docs/doxygen/GeosxConfig.hpp:33]: the edits here seem out of scope for this PR

**Review by joshua-white — COMMENTED** [review]
In general, this looks really great.  My two overall comments are:
1. We should revisit the data storage and see if we can lighten it by dropping any temp arrays that could actually be computed on the fly.  Not being completely familiar with the algorithm, perhaps all of these are essential, but perhaps not.
2. Now that you have a working version, perhaps we can revisit some of the variable names, just to make sure we like the notation and it clearly conveys the variable's nature and purpose.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.hpp (hunk 15 above)
[the remark this answers is not in the document]
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.hpp:49]: Hi Josh, the model only assumes a single slip direction so only two are required. Although there's an infinite number of shear directions given a normal vector and two orthogonal ones for Mode II and Mode III shear, the model is a single-degree-of-freedom system and doesn't differentiate between the two.

File: src/coreComponents/unitTests/inducedSeismicityTests/testDieterichSeismicityRateIntegralSolver.cpp [hunk 18]
```diff
@@ -0,0 +1,287 @@
...
+    <Mesh>
+      <InternalMesh
+        name="mesh"
+        elementTypes="{ C3D8 }"
+        xCoords="{ 0, 1e3 }"
+        yCoords="{ 0, 1e3 }"
+        zCoords="{ 0, 1e3 }"
+        nx="{ 10 }"
+        ny="{ 10 }"
+        nz="{ 10 }"
```
[the remark this answers is not in the document]
Comment (kimtaeho07) [reply src/coreComponents/unitTests/inducedSeismicityTests/testDieterichSeismicityRateIntegralSolver.cpp:68]: A 2D example would work fine for the purpose of this test. I'll change the mesh and move this to integrated tests in addition to writing a poroelastic example there.

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp (hunk 12 above)
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateBase.cpp]: I think this is a good idea. It would require solving for the optimal orientation given the current stress tensor. We could mark it as an issue.

File: src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp (hunk 16 above)
Comment (kimtaeho07) [reply src/coreComponents/physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp:31]: Deleted defunct fields and one of the _n variables that is now updated on the fly

File: src/coreComponents/unitTests/inducedSeismicityTests/testDieterichSeismicityRateIntegralSolver.cpp (hunk 18 above)
[the remark this answers is not in the document]
Comment (kimtaeho07) [reply src/coreComponents/unitTests/inducedSeismicityTests/testDieterichSeismicityRateIntegralSolver.cpp:68]: Per discussion with Herve, leaving as unit test for now

File: inputFiles/inducedSeismicity/faultMesh_2D_irreg.vtk
Comment (kimtaeho07) [inline inputFiles/inducedSeismicity/faultMesh_2D_irreg.vtk:1]: permeableFaultPoro.xml - poromechanical problem without fracture on normal (non-fractured) mesh
permeableFaultPoro_fractured.xml - same problem but with fracture on fractured mesh. Only converges with high newtonTol ~ 1e-1 due to high residual in solid solver
fault_fractured.xml - lagrangian contact solver, gives weird error about the constitutiveModels of fracture elementSubRegion having no children

**Review by kimtaeho07 — COMMENTED** [review]
.

File: src/.vscode/settings.json
Comment (CusiniM) [inline src/.vscode/settings.json:1]: please don't commit this file.

**Review by herve-gross — COMMENTED** [review]
The code look good, I have a few questions on the PR itself: should we include all the vtk files (vtm, vtp, vtu)? Also, there may be merge conflicts with the dedicated well skin factor implementation PR. This PR seems to encompass the skin factor PR.

**Comment by CusiniM** [comment]
> The code look good, I have a few questions on the PR itself: should we include all the vtk files (vtm, vtp, vtu)? Also, there may be merge conflicts with the dedicated well skin factor implementation PR. This PR seems to encompass the skin factor PR.

We should certainly remove most vtk, vtm, ... files and move them to the data repository. 

I am not too worried about the conflicts related to the well skin factor stuff. We can just checkout whatever comes from develop when merging develop into this branch.

But yeah, there is a little clean up to do.

File: src/coreComponents/python/modules/geosx_mesh_doctor/tests/.vscode/launch.json
Comment (TotoGaz) [inline src/coreComponents/python/modules/geosx_mesh_doctor/tests/.vscode/launch.json:1]: Please remove

File: src/coreComponents/python/modules/geosx_mesh_doctor/checks/generate_fractures.py
Comment (TotoGaz) [inline src/coreComponents/python/modules/geosx_mesh_doctor/checks/generate_fractures.py:1]: Can you remove this file since the feature has been 100% rewritten?

File: src/.vscode/settings.json
Comment (TotoGaz) [inline src/.vscode/settings.json:1]: Please remove

File: inputFiles/inducedSeismicity/fault_fractured.pvd
Comment (TotoGaz) [inline inputFiles/inducedSeismicity/fault_fractured.pvd:1]: The output is plain flat along the inputs? I think they should be moved to the baselines/integratedTests folder/repo?

File: src/coreComponents/mesh/generators/VTKFaceBlockUtilities.cpp [hunk 19]
```diff
@@ -223,7 +223,7 @@ class DuplicatedNodes
     // Field data key for duplicated nodes.
     constexpr char key[] = "duplicated_nodes";
 
-    vtkIntArray const * duplicatedNodes = vtkIntArray::FastDownCast( faceMesh->GetPointData()->GetArray( key ) );
+    vtkIdTypeArray const * duplicatedNodes = vtkIdTypeArray::FastDownCast( faceMesh->GetPointData()->GetArray( key ) );
     GEOS_ERROR_IF( duplicatedNodes == nullptr, "Could not find valid field \"" << key << "\" for fracture \"" << faceBlockName << "\"." );
```
Comment (TotoGaz) [inline src/coreComponents/mesh/generators/VTKFaceBlockUtilities.cpp]: Can you change the key to `colloctated_nodes` and the variable to `collocatedNodes` as well please?

File: src/coreComponents/unitTests/CMakeLists.txt [hunk 20]
```diff
@@ -11,4 +11,9 @@ add_subdirectory( finiteVolumeTests )
 add_subdirectory( fileIOTests )
 add_subdirectory( fluidFlowTests )
 add_subdirectory( wellsTests )
+<<<<<<< HEAD
+add_subdirectory( wavePropagationTests ) 
+add_subdirectory( inducedSeismicityTests ) 
+=======
 add_subdirectory( wavePropagationTests )
+>>>>>>> origin/develop
```
Comment (CusiniM) [inline src/coreComponents/unitTests/CMakeLists.txt]:
```suggestion
add_subdirectory( wavePropagationTests ) 
add_subdirectory( inducedSeismicityTests ) 
```

File: .gitmodules [hunk 21]
```diff
@@ -15,4 +15,4 @@
 	url = ../../GEOS-DEV/hdf5_interface.git
 [submodule "scripts/uberenv"]
 	path = scripts/uberenv
-	url = ../../LLNL/uberenv.git
+	url = https://github.com/LLNL/uberenv.git
```
Comment (CusiniM) [inline .gitmodules]:
```suggestion
	url = ../../LLNL/uberenv.git
```

File: src/docs/doxygen/GeosxConfig.hpp
Comment (CusiniM) [inline src/docs/doxygen/GeosxConfig.hpp:1]: need to revert this!!!

File: src/coreComponents/unitTests/inducedSeismicityTests/testSeismicityRateIntegralSolver.cpp
Comment (CusiniM) [inline src/coreComponents/unitTests/inducedSeismicityTests/testSeismicityRateIntegralSolver.cpp:1]: I am trying to get rid of this unitTest(it includes half of GEOS!) to replace it with an integratedTests with a curvecheck. Hopefully I ll succeed.

**Review by CusiniM — APPROVED** [review]

File: src/coreComponents/mesh/Perforation.hpp [hunk 22]
```diff
@@ -90,8 +90,9 @@ class Perforation : public dataRepository::Group
   real64 const & getDistanceFromWellHead() const { return m_distanceFromHead; }
 
   /**
-   * @brief Get the well Peaceman index at the perforation.
-   * @return the well transmissibility
+   * @class Perforation
+   *
+   * This class describes a perforation with its location, well transmissibility  and corresponding well element
```
Comment (CusiniM) [inline src/coreComponents/mesh/Perforation.hpp]:
```suggestion
   * @brief Get the well Peaceman index at the perforation.
   * @return the well transmissibility
```

File: src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateKernels.hpp [hunk 23]
```diff
@@ -0,0 +1,201 @@
...
+#include "physicsSolvers/inducedSeismicity/inducedSeismicityFields.hpp"
+#include "physicsSolvers/fluidFlow/FlowSolverBaseFields.hpp"
+
+namespace geos
+{
+
+namespace seismicityRateKernels
+{
+/**
+ * @class DieterichSeismicityRate
```
Comment (CusiniM) [inline src/coreComponents/physicsSolvers/inducedSeismicity/SeismicityRateKernels.hpp:29]:
```suggestion
 * @class SeismicityRateKernel
```

File: inputFiles/inducedSeismicity/inducedSeismicity.ats [hunk 24]
```diff
@@ -0,0 +1,27 @@
...
+
+decks = [
+    TestDeck(
+        name="SeismicityRate_1d_verification_smoke",
+        description="",
+        partitions=((1, 1, 1), (2, 1, 1)),
+        restart_step=10,
+        check_step=20,
+        restartcheck_params=RestartcheckParameters(atol=1e-4, rtol=1e-3)),
+        TestDeck(
```
Comment (CusiniM) [inline inputFiles/inducedSeismicity/inducedSeismicity.ats]:
```suggestion
     TestDeck(
```

File: src/docs/doxygen/GeosxConfig.hpp [hunk 25]
```diff
@@ -68,6 +68,18 @@
...
+  /// Parsed hypre version information
+  #define HYPRE_VERSION_MAJOR 2
+  /// Parsed hypre version information
+  #define HYPRE_VERSION_MINOR 28
+  /// Parsed hypre version information
+  #define HYPRE_VERSION_PATCH 0
+#endif
+
+=======
+>>>>>>> origin/develop
```
Comment (CusiniM) [inline src/docs/doxygen/GeosxConfig.hpp]:
```suggestion
```

File: src/docs/doxygen/GeosxConfig.hpp [hunk 26]
```diff
@@ -171,7 +183,11 @@
 #define fmt_VERSION 10.0.0
 
 /// Version information for python
+<<<<<<< HEAD
+#define Python3_VERSION 3.10.12
+=======
```
Comment (CusiniM) [inline src/docs/doxygen/GeosxConfig.hpp]:
```suggestion
```

File: src/docs/doxygen/GeosxConfig.hpp [hunk 27]
```diff
@@ -171,7 +183,11 @@
...
 #define Python3_VERSION 3.10.6
+>>>>>>> origin/develop
```
Comment (CusiniM) [inline src/docs/doxygen/GeosxConfig.hpp]:
```suggestion
```
